### DISCUSSION TITLE
Feature Request: Add Table Widget to Elementor

### DISCUSSION THREAD

**Discussion by ayoubkhan558** [elementor/elementor#28133] [Feature Request]
### Prerequisites

- [X] I have searched for similar feature requests in both open and closed discussions and cannot find an existing request.
- [X] I have verified that the feature is still missing from the latest stable versions of Elementor or Elementor Pro.

### What problem is your feature request going to solve? Please describe..

The absence of a dedicated table widget in Elementor limits users who need to display tabular data. This functionality is essential for presenting data in an organized and accessible manner, such as for product comparisons, pricing tables, schedules, and data reports.

![image](https://github.com/user-attachments/assets/339aa63f-3b15-48a3-a384-d02a366904c2)

### Describe the solution you'd like.

A dedicated Table widget in Elementor that allows users to easily create and customize tables. The widget should support:

- Adding, editing, and deleting rows and columns.
- Merging cells for advanced table structures.
- Customizing the appearance of the table, including borders, background colors, text alignment, and padding.
- Adding dynamic content using custom fields (e.g., ACF) or posts.

### Describe alternatives you've considered.

Currently, users must rely on external plugins or manually create tables using HTML and CSS within the Text Editor widget. Both alternatives are less user-friendly and do not integrate seamlessly with Elementor's interface.

https://powerpackelements.com/elementor-widgets/table/
https://crocoblock.com/widgets/table/
https://wpmet.com/plugin/elementskit/table/
https://wordpress.org/plugins/table-addons-for-elementor/

### Additional context

Providing a native Table widget would enhance Elementor's flexibility and user experience. It aligns with Elementor's goal of offering comprehensive tools for website building without relying on third-party plugins.

Key features of the widget:

- Rows and Columns Management: Easily add, remove, and rearrange rows and columns.
- Cell Options: Merge cells, set cell content alignment, and customize cell padding.
- Styling: Customize table borders, cell backgrounds, text colors, and font sizes.
- Dynamic Content Integration: Support for dynamic content from custom fields, posts, or other sources.
- Responsive Design: Ensure the table displays correctly on different devices, with options to hide/show certain columns or rows on mobile devices.

### Agreement

- [X] I agree that my request may be closed without action if it doesn't meet all the requirements.

**Comment by Maxime-JULIEN68** [comment]
**My Custom Table Widget – A Solution for the Missing Features in Existing Plugins**
I've encountered the same issue—there are no truly suitable plugin options available. I've tested several plugins and addons, including Royal Addons, which is great overall. However, I've run into some issues with JavaScript libraries that prevent me from achieving exactly what I need.

So, I decided to create a **custom Elementor widget using AI**, taking inspiration from the Royal Addons table widget but adding extra features that better suit my needs.

I've attached the result in the ZIP file. This custom widget introduces several new customization options, including:
✅ **Sticky or non-sticky thead** – You can choose whether the table header stays fixed while scrolling.
✅ **Customizable max height & max width** – Adjust the table’s dimensions easily.
✅ **Image support in table cells** – Add images directly inside the table, making it more flexible for design needs.
✅ **Well-structured system for adding cells & rows** – The method for adding columns (thead) and rows (tbody) is separate and structured for better usability.
✅ **Improved parent div controls** – You can customize border-radius and shadow, enhancing the table’s visual appearance.
✅ **Fine-tuned design options** – Additional small but useful settings make table customization much more practical.

🔧 **What Needs Improvement?**
There are still a few things that need refinement:
🔹 The row and cell-adding functions need verification to ensure full functionality.
🔹 Some features aren’t working as expected and require debugging.
🔹 Certain UI controls (like color variables in the content panel) need optimization.
🔹 More improvements could be made to enhance user experience and flexibility.

Despite these remaining issues, the core functionality is already in place, and this widget could be a great alternative for users who need more flexibility in their Elementor tables.

I hope this contribution helps others facing the same challenges! Let me know if anyone is interested in improving it further.

[Plugin table personnalise 9.2.3.zip](https://github.com/user-attachments/files/18905520/Plugin.table.personnalise.9.2.3.zip)

**Comment by tehben** [comment]
And responsive design could contain options for scroll or data arrangement as per WP Data Tables / TablePress
